migration/sql-files/062-workitem-related-changes.sql

```diff
@@ -0,0 +1,77 @@
+-- add a column to record the timestamp of the latest addition/change/removal of a comment on a workitem
+ALTER TABLE work_items ADD COLUMN commented_at timestamp with time zone;
+
+-- trigger to fill the `commented_at` column when a comment is added, updated or removed (soft delete, it's a record update)
+CREATE FUNCTION workitem_comment_insert_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.created_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION workitem_comment_update_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.updated_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION workitem_comment_softdelete_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.deleted_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER workitem_comment_insert_trigger AFTER INSERT ON comments 
```

Comment:
@xcoulon  Not directly related to comments but in the old implementation of space templates (and probably in the new one) I had to create a space in soft deleted state in order to apply a space template to it. This was some sort of *live verification* that a template works.